fastestimator/pipeline.py

```diff
@@ -112,43 +114,47 @@ def benchmark(self, mode: str = "train", epoch: int = 0, num_steps: int = 1000,
             loader = loader.take(num_steps)
         start = time.perf_counter()
         for idx, _ in enumerate(loader):
-            if idx == num_steps:
-                break
-            if idx % log_interval == 0 and idx > 0:
+            if idx + 1 % log_interval == 0:
                 duration = time.perf_counter() - start
                 iters_per_sec = log_interval / duration
-                print("FastEstimator: Step: {}, Epoch: {}, Steps/sec: {}".format(idx, epoch, iters_per_sec))
+                print("FastEstimator: Step: {}, Epoch: {}, Steps/sec: {}".format(idx + 1, epoch, iters_per_sec))
                 start = time.perf_counter()
+            if idx == num_steps:
+                break
 
-    def transform(self, data: Dict[str, Any], mode: str, epoch: int = 0) -> Dict[str, Any]:
+    def transform(self, data: Dict[str, Any], mode: str, epoch: int = 1) -> Dict[str, Any]:
         """apply all pipeline operations on given data for certain mode and epoch.
 
         Args:
             data: Input data in dictionary format
             mode: Current mode, can be "train", "eval", "test" or "infer"
-            epoch: Current epoch index. Defaults to 0.
+            epoch: Current epoch index. Defaults to 1.
 
         Returns:
             transformed data
         """
+        data = deepcopy(data)
         ops = get_current_ops(self.ops, mode, epoch)
```

Comment:
we should abstract this into a single function somewhere so that pipeline and op dataset don't accidentally get out of sync again

Reply:
`pipeline.transform` and `network._forward_batch` and `opdataset._forward` are all very different. doesn't make much sense to wrap them.

Reply:
done